src/main/java/fi/vrk/xroad/monitor/actor/Supervisor.java

```diff
@@ -107,19 +117,27 @@ public Receive createReceive() {
     private void handleMonitorDataRequest(StartCollectingMonitorDataCommand request) {
         Timeout timeout = new Timeout(1, TimeUnit.MINUTES);
         try {
+            // init result collector with expected results
             Await.ready(Patterns.ask(resultCollectorActor, request.getSecurityServerInfos(), timeout),
                 timeout.duration());
-
-            request.getSecurityServerInfos().stream()
-                    .forEach(info -> {
-                        log.info("Process SecurityServerInfo {}", info);
-                        monitorDataRequestPoolRouter.tell(new MonitorDataHandlerActor.MonitorDataRequest(info),
-                            getSelf());
-                    });
-
         } catch (TimeoutException | InterruptedException e) {
             log.error("Failed to initialize the ResultCollectorActor, {}", e);
         }
+
+        try {
+            // init Elasticsearch
+            Await.ready(Patterns.ask(elasticsearchInitializerActor, "init", timeout),
+                timeout.duration());
+        } catch (TimeoutException | InterruptedException e) {
+            log.error("Failed to initialize the ElasticsearchInitializerActor, {}", e);
+        }
+
```

Comment:
Is it really safe to continue processing if the initialization has failed? 
Using blocking operations within an actor should be avoided. Consider refactoring to async.